internal/xds/translator/tcpkeepalive.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+)
+
+// buildTCPKeepaliveSocketOptions converts listener downstream tcpKeepAlive settings to xds socketOptions
+func buildTCPKeepaliveSocketOptions(keepAlive *ir.TCPKeepalive) []*corev3.SocketOption {
+	if keepAlive == nil {
+		return nil
+	}
+
+	socketOptions := make([]*corev3.SocketOption, 0)
+	// Enable Keep Alives
+	socketOption := &corev3.SocketOption{
+		Description: "socket option to enable tcp keep alive",
+		Level:       0x1,                                        // syscall.SOL_SOCKET has a different value for Darwin, resulting in `go test` failing
```

Comment:
do you means `go test` will failing when someone coding with mac?

Reply:
yes, this is from my macbook
```
git diff
diff --git a/internal/xds/translator/tcpkeepalive.go b/internal/xds/translator/tcpkeepalive.go
index ebf129ca..07201803 100644
--- a/internal/xds/translator/tcpkeepalive.go
+++ b/internal/xds/translator/tcpkeepalive.go
@@ -6,6 +6,8 @@
 package translator
 
 import (
+       "syscall"
+
        corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 
        "github.com/envoyproxy/gateway/internal/ir"
@@ -21,8 +23,8 @@ func buildTCPKeepaliveSocketOptions(keepAlive *ir.TCPKeepalive) []*corev3.Socket
        // Enable Keep Alives
        socketOption := &corev3.SocketOption{
                Description: "socket option to enable tcp keep alive",
-               Level:       0x1,                                        // syscall.SOL_SOCKET has a different value for Darwin, resulting in `go test` failing
-               Name:        0x9,                                        // syscall.SO_KEEPALIVE has a different value for Darwin, resulting in `go test` failing
+               Level:       syscall.SOL_SOCKET,                         // syscall.SOL_SOCKET has a different value for Darwin, resulting in `go test` failing
+               Name:        syscall.SO_KEEPALIVE,                       // syscall.SO_KEEPALIVE has a different value for Darwin, resulting in `go test` failing
                Value:       &corev3.SocketOption_IntValue{IntValue: 1}, // Enable
        }
 
🐳 ~/go-workspace/src/github.com/envoyproxy/gateway/internal/xds/translator$ make test^C
🐳 ~/go-workspace/src/github.com/envoyproxy/gateway/internal/xds/translator$ go test ./...
--- FAIL: TestTranslateXds (0.05s)
    --- FAIL: TestTranslateXds/listener-tcp-keepalive (0.00s)
        translator_test.go:218: 
            	Error Trace:	/Users/arkodebdasgupta/go-workspace/src/github.com/envoyproxy/gateway/internal/xds/translator/translator_test.go:218
            	Error:      	Not equal: 

            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -36,4 +36,4 @@
            	            	     intValue: "1"
            	            	-    level: "1"
            	            	-    name: "9"
            	            	+    level: "65535"
            	            	+    name: "8"
            	            	 - address:
            	            	@@ -74,4 +74,4 @@
            	            	     intValue: "1"
            	            	-    level: "1"
            	            	-    name: "9"
            	            	+    level: "65535"
            	            	+    name: "8"
            	            	   - description: socket option for keep alive probes
            	            	@@ -111,4 +111,4 @@
            	            	     intValue: "1"
            	            	-    level: "1"
            	            	-    name: "9"
            	            	+    level: "65535"
            	            	+    name: "8"
            	            	 - address:
            	            	@@ -129,4 +129,4 @@
            	            	     intValue: "1"
            	            	-    level: "1"
            	            	-    name: "9"
            	            	+    level: "65535"
            	            	+    name: "8"
            	            	   - description: socket option for keep alive probes
            	Test:       	TestTranslateXds/listener-tcp-keepalive
FAIL
FAIL	github.com/envoyproxy/gateway/internal/xds/translator	0.562s
ok  	github.com/envoyproxy/gateway/internal/xds/translator/runner	1.118s
FAIL
```

Reply:
prefer to make this const, and have detail comment on the top on this file.

Reply:
But the const won't work, as shared in above test

Reply:
I will take a look at later, this should not be a blocker.